crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -1880,6 +1883,12 @@ where
                         }
                         BeaconEngineMessage::EventListener(tx) => this.push_listener(tx),
                     }
+                }
+
+                // We don't drain the messages right away, because we want to sneak a polling of
+                // running hook in between them.
+                if let Poll::Ready(Some(msg)) = this.engine_message_stream.poll_next_unpin(cx) {
+                    this.queued_engine_messages.push_back(msg);
                     continue
                 }
```

Comment:
I think here we want to fully drain the stream to ensure we're woken up again and it doesn't matter if we keep them buffered in the channel or in the vecdeque

Reply:
alright, wasn't sure cause of the code comment